src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs

```diff
@@ -272,6 +272,21 @@ public void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
             }
         }
 
+        public void ChangeDefaultCredentialsPolicy(SafeWinHttpHandle requestHandle, bool allowDefaultCredentials)
+        {
+            uint optionData = allowDefaultCredentials ?
+                Interop.WinHttp.WINHTTP_AUTOLOGON_SECURITY_LEVEL_LOW :
```

Comment:
https://msdn.microsoft.com/en-us/library/windows/desktop/aa383144(v=vs.85).aspx#auto_logon

> The auto-logon policy was implemented to prevent these credentials from being casually used to authenticate against an untrusted server. By default, the security level is set to WINHTTP_AUTOLOGON_SECURITY_LEVEL_MEDIUM, which allows the default credentials to be used only for Intranet requests. The auto-logon policy only applies to the NTLM and Negotiate authentication schemes. Credentials are never automatically transmitted with other schemes.
[...]
When the policy is set to WINHTTP_AUTOLOGON_SECURITY_LEVEL_HIGH, default credentials cannot be used for authentication. It is __strongly recommended that you use the auto-logon at the MEDIUM level__.